File: Day1/day1.go
package day1

import "fmt"

func main() {
	test := []int {1721, 979, 366, 299, 675, 1456}



    fmt.Println("test: ")
    fmt.Println(computePart2(test))
    fmt.Println("Expecting 241861950")
}


func computePart2(input []int) int {
    for i, iv := range input {
        for  j := i + 1; j < len(input); j++ {
            for  k := j + 1; k < len(input); k++ {
                if (input[i] + input[j] + input[k] == 2020) {
                    fmt.Println("wining numbers", iv, input[j], input[k])
                    return iv * input[j] * input[k]
                }
            }
        }
	}
	return -1
}

File: Day1/day1.ts
let testInput = [
    1721,
    979,
    366,
    299,
    675,
    1456,
]

function computeDay1() {
    for (let i = 0; i < testInput.length; i++) {
        for (let j = i + 1; j < testInput.length; j++) {
            if (testInput[i] + testInput[j] == 2020) {
                console.log("wining numbers", testInput[i], testInput[j])
                return testInput[i] * testInput[j]
            }
        }
    }
}

function computePart2(input) {
    for (let i = 0; i < testInput.length; i++) {
        for (let j = i + 1; j < testInput.length; j++) {
            for (let k = j + 1; k < testInput.length; k++) {
                if (testInput[i] + testInput[j] + testInput[k] == 2020) {
                    console.log("wining numbers", testInput[i], testInput[j], testInput[k])
                    return testInput[i] * testInput[j] * testInput[k]
                }
            }
        }
    }
}

console.log(computeDay1())
